src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java

```diff
@@ -78,4 +109,25 @@ public String getTooltipText() {
             }
         };
     }
+
+    private void update(ServiceLevel serviceLevel) {
+        this.serviceLevel = serviceLevel;
```

Comment:
I was about to write a question on why running this if `this.serviceLevel == serviceLevel` and then I returned to the service class and saw the event will not be triggered in such case.
Can you add comment here about it to make it clearer?

Reply:
Added.